Features/login

vjacquet opened the pull request:
Fixed #1065, added support for client validation of required field and added sr-only labels for the input fields.

tidyui commented:
Hi there! I’m having a hard time following the changes in this PR as:

1. You don’t seem to have the same spaces/tabs setting which makes it impossible to diff the file.

2. Im pretty sure the feature you added was already added in 8.1.

We need to keep spaces/tabs consistent throughout the codebase in order to follow the changes being made.

Best regards

vjacquet commented:
Sorry about this, I was careful on the .cs file (I had a warning in the PR Quality Review for not using curly braces around a one line if statement which I fixed) but I overlooked the .cshtml file. 
Maybe you could add a .editorconfig file at the root of the solution to ensure consistency? VS2019 now a feature to generate it from your settings (https://docs.microsoft.com/en-us/visualstudio/ide/code-styles-and-code-cleanup?view=vs-2019#code-styles-in-editorconfig-files).

What I did in the .cs file was:
- rewrite the LoginModel class using Identity/Login as a template to support returnUrl
- remove the property SignInFailed to rely on the ModelState

What I did in the .cshtml file was:
- remove the script section (there is no layout so the section is never rendered) and added the autofocus on the username input field instead
- support bootstrap validation: the invalid-feedback element were there but the setup to use them was missing
- add the required attribute on the fields to validate them on the client side.
- add localization for the invalid-feedback element.

What I did in the .scss file was:
- fixed the styles to support invalid-feedback and was-validated
- rename $red-bright to $login-invalid-feedback-color to state the purpose of the variable
- changed the background color of the page to #0C3C50, closer to the background-image in order to limit the flickering effect.

vjacquet commented:
Hello,
I do not understand why LGTM fails. is there something I have to fix in my branch?
Best regards

tidyui commented:
Sorry for the delay! We haven’t had time to review the added parts en UI yet but we’re working on it!

vjacquet commented:
I admit I got a carried away following the boy scout rule 😅.

tidyui commented on core/Piranha.Manager/assets/src/scss/inc/_variables.scss:
```diff
@@ -17,7 +17,6 @@ $indigo:        #6610f2;
 $purple:        #6f42c1;
 $pink:          #e83e8c;
 $red:           #a94441;
-$red-bright:    #f94c3b;
```
This should not be renamed. This color is used for validation purposes in other parts of the manager and just not the login.

vjacquet replied:
I though I checked for any other occurences of this variable in the code. My mistake.

tidyui commented on core/Piranha.Manager/assets/src/scss/inc/_variables.scss:
```diff
@@ -93,3 +92,7 @@ $nav-tabs-inverted-link-active-border-color: $nav-tabs-inverted-link-active-bg $
+
+// Login
+
+$login-invalid-feedback-color: #f94c3b;
```
See above comment. This change should be removed

vjacquet replied:
Just did :)

tidyui commented:
@filipjansson We have client side validation in other parts of the manager. When these fail the input field is marked with a red border, why isn't this happening here? I was only partly involved with the validation implementation so I don't remember how it's set up 😀

vjacquet commented:
It is because the border of the .form-control was set to 0
https://github.com/PiranhaCMS/piranha.core/blob/f61b756bc4a8fb5420192ee5903ede23e0704e6f/core/Piranha.Manager/assets/src/scss/inc/_login.scss#L108

I guess it can be fixed by rewriting the rule as .form-control:not(:invalid). I can try and fix it in this branch but issue #1098 prevent me from checking the result.

vjacquet commented:
I was able to check the fix for the border thanks to @j2ba.
